src/ttauri/required.hpp

```diff
@@ -42,4 +42,7 @@ constexpr ssize_t operator"" _z(unsigned long long lhs) noexcept
     return static_cast<ssize_t>(lhs);
 }
 
+#define tt_short_circuit_self_assignment(other)\
```

Comment:
After talking with @PinkFlufflyLlama we think it is better if we call this `tt_return_on_self_assignment`. If you would like to change this that would be great.

If not, I will still merge this and change it at a later time.

Reply:
Addressed in cb9d45a540e34629d3b276cc91fc50f2097c3e5d.